src/resolve/resolved-dns-dnssec.c

```diff
@@ -14,12 +14,27 @@
 #include "sort-util.h"
 #include "string-table.h"
 
-#if PREFER_OPENSSL && OPENSSL_VERSION_MAJOR >= 3
-#  pragma GCC diagnostic push
-#    pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#if PREFER_OPENSSL
+#  include <openssl/bn.h>
+#  include <openssl/rsa.h>
+#  include <openssl/ec.h>
+#  include <openssl/ecdsa.h>
+
+DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(EC_POINT*, EC_POINT_free, NULL);
```

Comment:
This duplicates all these definitions many times over, which defeats the point of the common helper module, so please drop this commit

Reply:
to me the helper module should provide the abstractions so those includes arent needed outside (sometime in the future). but i am not the maintainer, will do.